docker/Dockerfile.oneapi

```diff
@@ -13,7 +13,7 @@ ARG HDF5_PATCH=2
 ARG HDF5_FIX=
 ARG PETSC_VERSION=3.20.0
 ARG SLEPC_VERSION=3.20.0
-ARG PYBIND11_VERSION=2.11.1
+ARG NANOBIND_VERSION=1.7.0
```

Comment:
This, and similar ones in the remaining docker files, will need to be bumped upon the next nanobind release